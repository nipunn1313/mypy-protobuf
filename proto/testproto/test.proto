syntax = "proto2";

package test;

import "mypy_protobuf/extensions.proto";
import "testproto/inner/inner.proto";
import "testproto/nested/nested.proto";
import "testproto/nopackage.proto";
import "testproto/test3.proto";

// Outer Enum
enum OuterEnum {
  // FOO
  FOO = 1;
  // BAR
  BAR = 2;
}

// Naming conflicts!
enum NamingConflicts {
  Name = 1;
  Value = 2;
  keys = 3;
  values = 4;
  items = 5;
  // See https://github.com/protocolbuffers/protobuf/issues/8803
  // proto itself generates broken code when DESCRIPTOR is there
  // DESCRIPTOR = 8;
}

// Message with one of everything
message Simple1 {
  // Inner Enum
  enum InnerEnum {
    // INNER1
    INNER1 = 1;
    // INNER2
    INNER2 = 2;
  }
  message InnerMessage {}

  optional string a_string = 1;
  repeated string a_repeated_string = 2;
  optional bool a_boolean = 3;
  optional uint32 a_uint32 = 4;
  optional OuterEnum a_enum = 5;
  optional test3.OuterEnum a_external_enum = 6;

  optional inner.Inner a_inner = 7;
  optional test.nested_package.Nested a_nested = 12;

  optional InnerEnum inner_enum = 8;
  repeated InnerEnum rep_inner_enum = 9;
  optional InnerMessage inner_message = 10;
  repeated InnerMessage rep_inner_message = 11;

  optional NoPackage no_package = 13;
  optional test.nested_package.AnotherNested.NestedEnum nested_enum = 14;
  optional test.nested_package.AnotherNested.NestedMessage nested_message = 15;

  oneof a_oneof {
      string a_oneof_1 = 16;
      string a_oneof_2 = 17;
      Simple2 outer_message_in_oneof = 18;
      OuterEnum outer_enum_in_oneof = 19;
      InnerEnum inner_enum_in_oneof = 20;
  }

  optional uint32 user_id = 21 [(mypy_protobuf.casttype)="test/test_generated_mypy.UserId"];
  optional string email = 22 [(mypy_protobuf.casttype)="test/test_generated_mypy.Email"];
  map<uint32, string> email_by_uid = 23 [
    (mypy_protobuf.keytype)="test/test_generated_mypy.UserId",
    (mypy_protobuf.valuetype)="test/test_generated_mypy.Email"
  ];

  extensions 1000 to max;
}

message Simple2 {
  required string a_string = 1;

  extensions 1000 to max;
}

message Extensions1 {
  extend Simple1 {
    // ext
    optional Extensions1 ext = 1000;
  }

  optional string ext1_string = 1;
}

message Extensions2 {
  extend Simple1 {
    // foo
    optional Extensions2 foo = 2020;
  }

  optional bool flag = 1;
}

message None {
  optional int64 valid = 1;
}

message PythonReservedKeywords {
  enum finally {
      continue = 1;
      valid_in_finally = 2;
  }

  message lambda {
      optional int64 continue = 1;
      optional int64 valid = 2;
  }

  required int64 from = 1;
  optional Simple2 in = 2;
  optional finally is = 3;
  optional int64 for = 5;
  optional int64 try = 6;
  optional int64 def = 7;
  optional int64 nonlocal = 8;
  optional int64 while = 9;
  optional int64 and = 10;
  optional int64 del = 11;
  optional int64 global = 12;
  optional int64 not = 13;
  optional int64 with = 14;
  optional int64 as = 15;
  optional int64 elif = 16;
  optional int64 if = 17;
  optional int64 or = 18;
  optional int64 yield = 19;
  optional int64 assert = 20;
  optional int64 else = 21;
  optional int64 import = 22;
  optional int64 pass = 23;
  optional int64 break = 24;
  optional int64 except = 25;
  optional int64 raise = 26;
  optional int64 False = 27;
  optional int64 True = 29;
  optional int64 class = 30;

  // Test unreserved identifiers w/ reserved message names
  optional None none = 28;
  optional finally valid = 31;
}

// Do one with just one arg - to make sure it's syntactically correct
message PythonReservedKeywordsSmall {
  required int64 from = 1;
}

// Method name is reserved
service PythonReservedKeywordsService {
    // lambda
    rpc lambda(Simple1) returns (PythonReservedKeywords.lambda) {}
    // valid_method_name1
    rpc valid_method_name1(Simple1) returns (None) {}
    // valid_method_name2
    rpc valid_method_name2(Simple1) returns (PythonReservedKeywords.lambda) {}
}

// when service name itself is reserved - generated code was found to be invalid
// in protoc 3.17.3
//service global {
//    rpc Echo(Simple1) returns (Simple2) {}
//}

option py_generic_services = true;
service ATestService {
    rpc Echo(Simple1) returns (Simple2) {}
}

message SelfField {
    // Field self -> must generate an __init__ method w/ different name
    optional int64 self = 1;
}
