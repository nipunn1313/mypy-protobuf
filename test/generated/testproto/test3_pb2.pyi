"""
@generated by mypy-protobuf.  Do not edit manually!
isort:skip_file
package test3"""

import builtins
import collections.abc
import google.protobuf.descriptor
import google.protobuf.internal.containers
import google.protobuf.internal.enum_type_wrapper
import google.protobuf.message
import sys
import test.test_generated_mypy
import typing

if sys.version_info >= (3, 10):
    import typing as typing_extensions
else:
    import typing_extensions

DESCRIPTOR: google.protobuf.descriptor.FileDescriptor

class _OuterEnum:
    ValueType = typing.NewType("ValueType", builtins.int)
    V: typing_extensions.TypeAlias = ValueType

class _OuterEnumEnumTypeWrapper(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[_OuterEnum.ValueType], builtins.type):
    DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor
    UNKNOWN: _OuterEnum.ValueType  # 0
    FOO3: _OuterEnum.ValueType  # 1
    BAR3: _OuterEnum.ValueType  # 2

class OuterEnum(_OuterEnum, metaclass=_OuterEnumEnumTypeWrapper): ...

UNKNOWN: OuterEnum.ValueType  # 0
FOO3: OuterEnum.ValueType  # 1
BAR3: OuterEnum.ValueType  # 2
global___OuterEnum = OuterEnum

@typing_extensions.final
class OuterMessage3(google.protobuf.message.Message):
    DESCRIPTOR: google.protobuf.descriptor.Descriptor

    A_STRING_FIELD_NUMBER: builtins.int
    a_string: builtins.str
    def __init__(
        self,
        *,
        a_string: builtins.str = ...,
    ) -> None: ...
    def ClearField(self, field_name: typing_extensions.Literal["a_string", b"a_string"]) -> None: ...

global___OuterMessage3 = OuterMessage3

@typing_extensions.final
class SimpleProto3(google.protobuf.message.Message):
    DESCRIPTOR: google.protobuf.descriptor.Descriptor

    class _InnerEnum:
        ValueType = typing.NewType("ValueType", builtins.int)
        V: typing_extensions.TypeAlias = ValueType

    class _InnerEnumEnumTypeWrapper(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[SimpleProto3._InnerEnum.ValueType], builtins.type):
        DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor
        INNER1: SimpleProto3._InnerEnum.ValueType  # 0
        INNER2: SimpleProto3._InnerEnum.ValueType  # 1

    class InnerEnum(_InnerEnum, metaclass=_InnerEnumEnumTypeWrapper): ...
    INNER1: SimpleProto3.InnerEnum.ValueType  # 0
    INNER2: SimpleProto3.InnerEnum.ValueType  # 1

    @typing_extensions.final
    class MapScalarEntry(google.protobuf.message.Message):
        DESCRIPTOR: google.protobuf.descriptor.Descriptor

        KEY_FIELD_NUMBER: builtins.int
        VALUE_FIELD_NUMBER: builtins.int
        key: builtins.int
        value: builtins.str
        def __init__(
            self,
            *,
            key: builtins.int = ...,
            value: builtins.str = ...,
        ) -> None: ...
        def ClearField(self, field_name: typing_extensions.Literal["key", b"key", "value", b"value"]) -> None: ...

    @typing_extensions.final
    class MapMessageEntry(google.protobuf.message.Message):
        DESCRIPTOR: google.protobuf.descriptor.Descriptor

        KEY_FIELD_NUMBER: builtins.int
        VALUE_FIELD_NUMBER: builtins.int
        key: builtins.int
        @property
        def value(self) -> global___OuterMessage3: ...
        def __init__(
            self,
            *,
            key: builtins.int = ...,
            value: global___OuterMessage3 | None = ...,
        ) -> None: ...
        def HasField(self, field_name: typing_extensions.Literal["value", b"value"]) -> builtins.bool: ...
        def ClearField(self, field_name: typing_extensions.Literal["key", b"key", "value", b"value"]) -> None: ...

    @typing_extensions.final
    class EmailByUidEntry(google.protobuf.message.Message):
        DESCRIPTOR: google.protobuf.descriptor.Descriptor

        KEY_FIELD_NUMBER: builtins.int
        VALUE_FIELD_NUMBER: builtins.int
        key: builtins.int
        value: builtins.str
        def __init__(
            self,
            *,
            key: builtins.int = ...,
            value: builtins.str = ...,
        ) -> None: ...
        def ClearField(self, field_name: typing_extensions.Literal["key", b"key", "value", b"value"]) -> None: ...

    A_STRING_FIELD_NUMBER: builtins.int
    A_REPEATED_STRING_FIELD_NUMBER: builtins.int
    A_OUTER_ENUM_FIELD_NUMBER: builtins.int
    OUTER_MESSAGE_FIELD_NUMBER: builtins.int
    INNER_ENUM_FIELD_NUMBER: builtins.int
    A_ONEOF_1_FIELD_NUMBER: builtins.int
    A_ONEOF_2_FIELD_NUMBER: builtins.int
    OUTER_MESSAGE_IN_ONEOF_FIELD_NUMBER: builtins.int
    OUTER_ENUM_IN_ONEOF_FIELD_NUMBER: builtins.int
    INNER_ENUM_IN_ONEOF_FIELD_NUMBER: builtins.int
    B_ONEOF_1_FIELD_NUMBER: builtins.int
    B_ONEOF_2_FIELD_NUMBER: builtins.int
    BOOL_FIELD_NUMBER: builtins.int
    OUTERENUM_FIELD_NUMBER: builtins.int
    OUTERMESSAGE3_FIELD_NUMBER: builtins.int
    MAP_SCALAR_FIELD_NUMBER: builtins.int
    MAP_MESSAGE_FIELD_NUMBER: builtins.int
    AN_OPTIONAL_STRING_FIELD_NUMBER: builtins.int
    USER_ID_FIELD_NUMBER: builtins.int
    EMAIL_FIELD_NUMBER: builtins.int
    EMAIL_BY_UID_FIELD_NUMBER: builtins.int
    a_string: builtins.str
    @property
    def a_repeated_string(self) -> google.protobuf.internal.containers.RepeatedScalarFieldContainer[builtins.str]: ...
    a_outer_enum: global___OuterEnum.ValueType
    @property
    def outer_message(self) -> global___OuterMessage3: ...
    inner_enum: global___SimpleProto3.InnerEnum.ValueType
    a_oneof_1: builtins.str
    a_oneof_2: builtins.str
    @property
    def outer_message_in_oneof(self) -> global___OuterMessage3: ...
    outer_enum_in_oneof: global___OuterEnum.ValueType
    inner_enum_in_oneof: global___SimpleProto3.InnerEnum.ValueType
    b_oneof_1: builtins.str
    b_oneof_2: builtins.str
    @property
    def bool(self) -> global___OuterMessage3: ...
    OuterEnum: global___OuterEnum.ValueType
    """Test having fieldname match messagename"""
    @property
    def OuterMessage3(self) -> global___OuterMessage3: ...
    @property
    def map_scalar(self) -> google.protobuf.internal.containers.ScalarMap[builtins.int, builtins.str]:
        """Test generation of map"""

    @property
    def map_message(self) -> google.protobuf.internal.containers.MessageMap[builtins.int, global___OuterMessage3]: ...
    an_optional_string: builtins.str
    user_id: test.test_generated_mypy.UserId
    email: test.test_generated_mypy.Email
    @property
    def email_by_uid(self) -> google.protobuf.internal.containers.ScalarMap[test.test_generated_mypy.UserId, test.test_generated_mypy.Email]: ...
    def __init__(
        self,
        *,
        a_string: builtins.str = ...,
        a_repeated_string: collections.abc.Iterable[builtins.str] | None = ...,
        a_outer_enum: global___OuterEnum.ValueType = ...,
        outer_message: global___OuterMessage3 | None = ...,
        inner_enum: global___SimpleProto3.InnerEnum.ValueType = ...,
        a_oneof_1: builtins.str = ...,
        a_oneof_2: builtins.str = ...,
        outer_message_in_oneof: global___OuterMessage3 | None = ...,
        outer_enum_in_oneof: global___OuterEnum.ValueType = ...,
        inner_enum_in_oneof: global___SimpleProto3.InnerEnum.ValueType = ...,
        b_oneof_1: builtins.str = ...,
        b_oneof_2: builtins.str = ...,
        bool: global___OuterMessage3 | None = ...,
        OuterEnum: global___OuterEnum.ValueType = ...,
        OuterMessage3: global___OuterMessage3 | None = ...,
        map_scalar: collections.abc.Mapping[builtins.int, builtins.str] | None = ...,
        map_message: collections.abc.Mapping[builtins.int, global___OuterMessage3] | None = ...,
        an_optional_string: builtins.str | None = ...,
        user_id: test.test_generated_mypy.UserId = ...,
        email: test.test_generated_mypy.Email = ...,
        email_by_uid: collections.abc.Mapping[test.test_generated_mypy.UserId, test.test_generated_mypy.Email] | None = ...,
    ) -> None: ...
    def HasField(self, field_name: typing_extensions.Literal["OuterMessage3", b"OuterMessage3", "_an_optional_string", b"_an_optional_string", "a_oneof", b"a_oneof", "a_oneof_1", b"a_oneof_1", "a_oneof_2", b"a_oneof_2", "an_optional_string", b"an_optional_string", "b_oneof", b"b_oneof", "b_oneof_1", b"b_oneof_1", "b_oneof_2", b"b_oneof_2", "bool", b"bool", "inner_enum_in_oneof", b"inner_enum_in_oneof", "outer_enum_in_oneof", b"outer_enum_in_oneof", "outer_message", b"outer_message", "outer_message_in_oneof", b"outer_message_in_oneof"]) -> builtins.bool: ...
    def ClearField(self, field_name: typing_extensions.Literal["OuterEnum", b"OuterEnum", "OuterMessage3", b"OuterMessage3", "_an_optional_string", b"_an_optional_string", "a_oneof", b"a_oneof", "a_oneof_1", b"a_oneof_1", "a_oneof_2", b"a_oneof_2", "a_outer_enum", b"a_outer_enum", "a_repeated_string", b"a_repeated_string", "a_string", b"a_string", "an_optional_string", b"an_optional_string", "b_oneof", b"b_oneof", "b_oneof_1", b"b_oneof_1", "b_oneof_2", b"b_oneof_2", "bool", b"bool", "email", b"email", "email_by_uid", b"email_by_uid", "inner_enum", b"inner_enum", "inner_enum_in_oneof", b"inner_enum_in_oneof", "map_message", b"map_message", "map_scalar", b"map_scalar", "outer_enum_in_oneof", b"outer_enum_in_oneof", "outer_message", b"outer_message", "outer_message_in_oneof", b"outer_message_in_oneof", "user_id", b"user_id"]) -> None: ...
    @typing.overload
    def WhichOneof(self, oneof_group: typing_extensions.Literal["_an_optional_string", b"_an_optional_string"]) -> typing_extensions.Literal["an_optional_string"] | None: ...
    @typing.overload
    def WhichOneof(self, oneof_group: typing_extensions.Literal["a_oneof", b"a_oneof"]) -> typing_extensions.Literal["a_oneof_1", "a_oneof_2", "outer_message_in_oneof", "outer_enum_in_oneof", "inner_enum_in_oneof"] | None: ...
    @typing.overload
    def WhichOneof(self, oneof_group: typing_extensions.Literal["b_oneof", b"b_oneof"]) -> typing_extensions.Literal["b_oneof_1", "b_oneof_2"] | None: ...

global___SimpleProto3 = SimpleProto3
