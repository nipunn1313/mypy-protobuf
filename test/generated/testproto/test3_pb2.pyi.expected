"""
@generated by mypy-protobuf.  Do not edit manually!
isort:skip_file
"""
import builtins
import google.protobuf.descriptor
import google.protobuf.internal.containers
import google.protobuf.internal.enum_type_wrapper
import google.protobuf.message
import typing
import typing_extensions

DESCRIPTOR: google.protobuf.descriptor.FileDescriptor = ...

class OuterEnum(metaclass=_OuterEnum):
    V = typing.NewType('V', builtins.int)

global___OuterEnum = OuterEnum

UNKNOWN = OuterEnum.V(0)
FOO3 = OuterEnum.V(1)
BAR3 = OuterEnum.V(2)

class _OuterEnum(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[OuterEnum.V], builtins.type):
    DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor = ...
    UNKNOWN = OuterEnum.V(0)
    FOO3 = OuterEnum.V(1)
    BAR3 = OuterEnum.V(2)

class OuterMessage3(google.protobuf.message.Message):
    DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
    A_BOOL_FIELD_NUMBER: builtins.int
    a_bool: builtins.bool = ...

    def __init__(self,
        *,
        a_bool : builtins.bool = ...,
        ) -> None: ...
    def ClearField(self, field_name: typing_extensions.Literal[u"a_bool",b"a_bool"]) -> None: ...
global___OuterMessage3 = OuterMessage3

class SimpleProto3(google.protobuf.message.Message):
    DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
    class InnerEnum(metaclass=_InnerEnum):
        V = typing.NewType('V', builtins.int)

    INNER1 = SimpleProto3.InnerEnum.V(0)
    INNER2 = SimpleProto3.InnerEnum.V(1)

    class _InnerEnum(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[InnerEnum.V], builtins.type):
        DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor = ...
        INNER1 = SimpleProto3.InnerEnum.V(0)
        INNER2 = SimpleProto3.InnerEnum.V(1)

    class MapScalarEntry(google.protobuf.message.Message):
        DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
        KEY_FIELD_NUMBER: builtins.int
        VALUE_FIELD_NUMBER: builtins.int
        key: builtins.int = ...
        value: typing.Text = ...

        def __init__(self,
            *,
            key : builtins.int = ...,
            value : typing.Text = ...,
            ) -> None: ...
        def ClearField(self, field_name: typing_extensions.Literal[u"key",b"key",u"value",b"value"]) -> None: ...

    class MapMessageEntry(google.protobuf.message.Message):
        DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
        KEY_FIELD_NUMBER: builtins.int
        VALUE_FIELD_NUMBER: builtins.int
        key: builtins.int = ...

        @property
        def value(self) -> global___OuterMessage3: ...

        def __init__(self,
            *,
            key : builtins.int = ...,
            value : typing.Optional[global___OuterMessage3] = ...,
            ) -> None: ...
        def HasField(self, field_name: typing_extensions.Literal[u"value",b"value"]) -> builtins.bool: ...
        def ClearField(self, field_name: typing_extensions.Literal[u"key",b"key",u"value",b"value"]) -> None: ...

    A_STRING_FIELD_NUMBER: builtins.int
    A_REPEATED_STRING_FIELD_NUMBER: builtins.int
    A_OUTER_ENUM_FIELD_NUMBER: builtins.int
    OUTER_MESSAGE_FIELD_NUMBER: builtins.int
    INNER_ENUM_FIELD_NUMBER: builtins.int
    A_ONEOF_1_FIELD_NUMBER: builtins.int
    A_ONEOF_2_FIELD_NUMBER: builtins.int
    OUTER_MESSAGE_IN_ONEOF_FIELD_NUMBER: builtins.int
    OUTER_ENUM_IN_ONEOF_FIELD_NUMBER: builtins.int
    INNER_ENUM_IN_ONEOF_FIELD_NUMBER: builtins.int
    B_ONEOF_1_FIELD_NUMBER: builtins.int
    B_ONEOF_2_FIELD_NUMBER: builtins.int
    BOOL_FIELD_NUMBER: builtins.int
    OUTERENUM_FIELD_NUMBER: builtins.int
    OUTERMESSAGE3_FIELD_NUMBER: builtins.int
    MAP_SCALAR_FIELD_NUMBER: builtins.int
    MAP_MESSAGE_FIELD_NUMBER: builtins.int
    AN_OPTIONAL_STRING_FIELD_NUMBER: builtins.int
    a_string: typing.Text = ...

    @property
    def a_repeated_string(self) -> google.protobuf.internal.containers.RepeatedScalarFieldContainer[typing.Text]: ...
    a_outer_enum: global___OuterEnum.V = ...
    inner_enum: global___SimpleProto3.InnerEnum.V = ...
    a_oneof_1: typing.Text = ...
    a_oneof_2: typing.Text = ...
    outer_enum_in_oneof: global___OuterEnum.V = ...
    inner_enum_in_oneof: global___SimpleProto3.InnerEnum.V = ...
    b_oneof_1: typing.Text = ...
    b_oneof_2: typing.Text = ...
    OuterEnum: global___OuterEnum.V = ...
    an_optional_string: typing.Text = ...

    @property
    def outer_message(self) -> global___OuterMessage3: ...

    @property
    def outer_message_in_oneof(self) -> global___OuterMessage3: ...

    @property
    def bool(self) -> global___OuterMessage3: ...

    @property
    def OuterMessage3(self) -> global___OuterMessage3: ...

    @property
    def map_scalar(self) -> google.protobuf.internal.containers.ScalarMap[builtins.int, typing.Text]: ...

    @property
    def map_message(self) -> google.protobuf.internal.containers.MessageMap[builtins.int, global___OuterMessage3]: ...

    def __init__(self,
        *,
        a_string : typing.Text = ...,
        a_repeated_string : typing.Optional[typing.Iterable[typing.Text]] = ...,
        a_outer_enum : global___OuterEnum.V = ...,
        outer_message : typing.Optional[global___OuterMessage3] = ...,
        inner_enum : global___SimpleProto3.InnerEnum.V = ...,
        a_oneof_1 : typing.Text = ...,
        a_oneof_2 : typing.Text = ...,
        outer_message_in_oneof : typing.Optional[global___OuterMessage3] = ...,
        outer_enum_in_oneof : global___OuterEnum.V = ...,
        inner_enum_in_oneof : global___SimpleProto3.InnerEnum.V = ...,
        b_oneof_1 : typing.Text = ...,
        b_oneof_2 : typing.Text = ...,
        bool : typing.Optional[global___OuterMessage3] = ...,
        OuterEnum : global___OuterEnum.V = ...,
        OuterMessage3 : typing.Optional[global___OuterMessage3] = ...,
        map_scalar : typing.Optional[typing.Mapping[builtins.int, typing.Text]] = ...,
        map_message : typing.Optional[typing.Mapping[builtins.int, global___OuterMessage3]] = ...,
        an_optional_string : typing.Text = ...,
        ) -> None: ...
    def HasField(self, field_name: typing_extensions.Literal[u"OuterMessage3",b"OuterMessage3",u"_an_optional_string",b"_an_optional_string",u"a_oneof",b"a_oneof",u"a_oneof_1",b"a_oneof_1",u"a_oneof_2",b"a_oneof_2",u"an_optional_string",b"an_optional_string",u"b_oneof",b"b_oneof",u"b_oneof_1",b"b_oneof_1",u"b_oneof_2",b"b_oneof_2",u"bool",b"bool",u"inner_enum_in_oneof",b"inner_enum_in_oneof",u"outer_enum_in_oneof",b"outer_enum_in_oneof",u"outer_message",b"outer_message",u"outer_message_in_oneof",b"outer_message_in_oneof"]) -> builtins.bool: ...
    def ClearField(self, field_name: typing_extensions.Literal[u"OuterEnum",b"OuterEnum",u"OuterMessage3",b"OuterMessage3",u"_an_optional_string",b"_an_optional_string",u"a_oneof",b"a_oneof",u"a_oneof_1",b"a_oneof_1",u"a_oneof_2",b"a_oneof_2",u"a_outer_enum",b"a_outer_enum",u"a_repeated_string",b"a_repeated_string",u"a_string",b"a_string",u"an_optional_string",b"an_optional_string",u"b_oneof",b"b_oneof",u"b_oneof_1",b"b_oneof_1",u"b_oneof_2",b"b_oneof_2",u"bool",b"bool",u"inner_enum",b"inner_enum",u"inner_enum_in_oneof",b"inner_enum_in_oneof",u"map_message",b"map_message",u"map_scalar",b"map_scalar",u"outer_enum_in_oneof",b"outer_enum_in_oneof",u"outer_message",b"outer_message",u"outer_message_in_oneof",b"outer_message_in_oneof"]) -> None: ...
    @typing.overload
    def WhichOneof(self, oneof_group: typing_extensions.Literal[u"_an_optional_string",b"_an_optional_string"]) -> typing.Optional[typing_extensions.Literal["an_optional_string"]]: ...
    @typing.overload
    def WhichOneof(self, oneof_group: typing_extensions.Literal[u"a_oneof",b"a_oneof"]) -> typing.Optional[typing_extensions.Literal["a_oneof_1","a_oneof_2","outer_message_in_oneof","outer_enum_in_oneof","inner_enum_in_oneof"]]: ...
    @typing.overload
    def WhichOneof(self, oneof_group: typing_extensions.Literal[u"b_oneof",b"b_oneof"]) -> typing.Optional[typing_extensions.Literal["b_oneof_1","b_oneof_2"]]: ...
global___SimpleProto3 = SimpleProto3
